Source/WebCore/Modules/compression/DecompressionStream.h

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) 2022 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#pragma once
+
+#include "ActiveDOMObject.h"
+#include "Document.h"
+#include "Event.h"
+#include "EventTarget.h"
+#include "ExceptionOr.h"
+#include "ReadableStream.h"
+#include "ReadableStreamSource.h"
+#include "ScriptExecutionContext.h"
+#include "WritableStream.h"
+
+#include <wtf/IsoMalloc.h>
+#include <wtf/IsoMallocInlines.h>
+#include <wtf/RefCounted.h>
+#include <wtf/text/WTFString.h>
+
+namespace JSC {
+class CallFrame;
+class JSGlobalObject;
+class JSValue;
+}
+
+namespace WebCore {
+
+class DecompressionStream : public RefCounted<DecompressionStream>,
+    public ActiveDOMObject,
+    public EventTargetWithInlineData {
+    WTF_MAKE_ISO_ALLOCATED(DecompressionStream);
+
+public:
+    // Compression Formats
+    // gzip         "GZIP file format"              [RFC1952]
+    // deflate      "ZLIB Compressed Data Format"   [RFC1950]
+    // deflate-raw  "The DEFLATE algorithm"         [RFC1951]
+    enum CompressionFormat { Gzip, Deflate, Deflate_raw };
```

Comment:
This implies we will sometimes do:
`DecompressionStream::CompressionFormat::Gzip` and other times `CompressionStream::CompressionFormat::Gzip`, depending on the call site.

It seems like having a single enum for both cases would be better. But I'm not sure where it would be best to declare that enum -- perhaps in a file called "Modules/compression/formats.h" or something?

This should probably also be:
enum class CompressionFormat : uint8_t { Gzip, Deflate, Deflate_raw };

Reply:
Yes, I was thinking about doing this earlier. I wasn't sure if I just wanted to create an entire file just for one shared enum. I'll update the patch though with this new file.